services/QuillLMS/app/models/districts_admins.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: districts_admins
+#
+#  id          :bigint           not null, primary key
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  district_id :bigint           not null
+#  user_id     :bigint           not null
+#
+# Indexes
+#
+#  index_districts_admins_on_district_id  (district_id)
+#  index_districts_admins_on_user_id      (user_id)
+#
+class DistrictsAdmins < ApplicationRecord
+  belongs_to :district
+  belongs_to :user
+
+  after_create :attach_to_subscribed_schools
+  after_destroy :detach_from_schools
+
+  def admin
+    user
+  end
+
+  def attach_to_subscribed_schools
+    district = District.find(district_id)
+    schools_with_subscriptions.each do |school|
+      school_admin = SchoolsAdmins.find_or_create_by(school: school, user: admin)
+      NewAdminEmailWorker.perform_async(user_id, school.id)
+    end
```

Comment:
You can create the records without the loop, but I think you need the loop for the worker calls.  Maybe...
```ruby
admin.administered_schools += schools_with_subscriptions
admin.save
schools_with_subscriptions.each { |school| NewAdminEmailWorker.perform_async(user_id, school.id) }
```